WordPress/Classes/ViewRelated/Stats/JetpackLoginViewController.swift

```diff
@@ -0,0 +1,411 @@
+
+    @IBOutlet fileprivate weak var jetpackImage: UIImageView!
+    @IBOutlet fileprivate weak var descriptionLabel: UILabel!
+    @IBOutlet fileprivate weak var usernameTextField: WPWalkthroughTextField!
+    @IBOutlet fileprivate weak var passwordTextField: WPWalkthroughTextField!
+    @IBOutlet fileprivate weak var verificationCodeTextField: WPWalkthroughTextField!
+    @IBOutlet fileprivate weak var scrollView: UIScrollView!
+    @IBOutlet fileprivate weak var signinButton: WPNUXMainButton!
+
+    fileprivate var blog: Blog!
+    fileprivate var activeField: UITextField?
+    fileprivate var shouldDisplayMultifactor: Bool = false
+    fileprivate var loginFields = LoginFields()
+
+    fileprivate lazy var loginFacade: LoginFacade = {
+        let facade = LoginFacade()
+        facade.delegate = self
+        return facade
+    }()
+
+    /// Returns true if this VC is currently authenticating with the server.
+    /// After setting, the UI controls will be updated accordingly
+    ///
+    fileprivate var isAuthenticating: Bool = false {
+        didSet {
+            self.usernameTextField.isEnabled = !self.isAuthenticating
+            self.passwordTextField.isEnabled = !self.isAuthenticating
+            self.verificationCodeTextField.isEnabled = !self.isAuthenticating
+            self.signinButton.showActivityIndicator(self.isAuthenticating)
+        }
+    }
+
+    /// Returns true if the blog has the proper version of Jetpack installed,
+    /// otherwise false
+    ///
+    fileprivate var hasJetpack: Bool {
+        guard let jetpack = self.blog.jetpack else {
+            return false
+        }
+        return (jetpack.isInstalled() && jetpack.isUpdatedToRequiredVersion())
+    }
+
+    // MARK: - Initializers
+
+    /// Preferred initializer for JetpackLoginViewController
+    ///
+    /// - Parameter blog: The current blog
+    ///
+    convenience init(blog: Blog) {
+        self.init()
+        self.blog = blog
+    }
+
+    // MARK: - LifeCycle Methods
+
+    override func viewDidLoad() {
+        assert(self.blog != nil)
+        super.viewDidLoad()
+        self.view.backgroundColor = WPStyleGuide.itsEverywhereGrey()
+        setupControls()
+        setupKeyboard()
+    }
+
+    override func viewDidLayoutSubviews() {
+        reloadInterface()
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        registerForKeyboardNotifications()
+        registerForTextFieldNotifications()
+        prefillJetPackUsernameIfAvailible()
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        deregisterFromKeyboardNotifications()
+        deregisterFromTextFieldNotifications()
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
+
+    // MARK: - Configuration
+
+    func setupControls() {
+        self.passwordTextField.font = WPNUXUtility.descriptionTextFont()
```

Comment:
@catehstn There is indeed a xib. The font's and colors typically were contained in a class to make changing styles easier app wide. However most of this can be pushed there. I will clean it up!